routes/carts.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express');
+const routerCarts = express.Router();
+const knex = require('../database');
+
+routerCarts.route('/:user_id').get((req, res) => {
+  const id = req.url.slice(1);
+  knex
+    .raw('SELECT id FROM users WHERE id = ?', [id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return res.send(404, '{ "message": "User not found" }');
+      }
+      knex
+        .raw(
+          'SELECT products.* FROM users INNER JOIN carts ON carts.user_id = users.id INNER JOIN products ON products.id = carts.product_id WHERE users.id = ?',
+          [id],
+        )
+        .then((result) => {
+          res.send(result.rows);
+        })
+        .catch((err) => {
```

Comment:
So if you return this inner promise, you don't need this catch here, since any errors on this inner promise chain will propagate up to the outer promise chain. But it only works if you return the promise.